cmake/ProcessOptions.cmake

```diff
@@ -392,11 +392,15 @@ function( NEST_PROCESS_WITH_PYTHON )
   elseif ( ${with-python} STREQUAL "ON" )
 
     # Localize the Python interpreter and ABI
-    if ( ${CMAKE_VERSION} VERSION_LESS "3.18.0")
+    find_package( Python 3.8 QUIET COMPONENTS Interpreter Development.Module )
+    if ( NOT Python_FOUND )
       find_package( Python 3.8 REQUIRED Interpreter Development )
-      message( WARNING "CMake 3.18+ is recommended for more universal Python bindings. If you encounter missing Python header or library file errors, try upgrading CMake.")
-    else()
-      find_package( Python 3.8 REQUIRED Interpreter Development.Module )
+      string( CONCAT PYABI_WARN "Could not locate Python ABI"
+        ", using shared libraries and header file instead."
+        " Please clear your CMake cache and build folder and verify that CMake"
+        " is up-to-date (3.18+)."
+      )
+      message( WARNING "${PYABI_WARN}")
     endif()
 
     if ( Python_FOUND )
```

Comment:
What about so?

```suggestion
    else()
      find_package( Python 3.8 REQUIRED Interpreter Development.Module )
```

Reply:
Ah! no, this won't be right, wouldn't it? We need the additional if below and should add the else to the part above.